pkg/operator/baremetal_config.go

```diff
@@ -65,3 +77,89 @@ func getBaremetalProvisioningConfig(dc dynamic.Interface, configName string) (Ba
 		ProvisioningDHCPRange:    provisioningDHCPRange,
 	}, nil
 }
+
+func getProvisioningIPCIDR(baremetalConfig BaremetalProvisioningConfig) string {
+	if baremetalConfig.ProvisioningIp != "" {
+		cidr := strings.Split(baremetalConfig.ProvisioningNetworkCIDR, "/")
+		return baremetalConfig.ProvisioningIp + "/" + cidr[1]
+	} else {
+		return unknownProvisioningConfig
+	}
+}
+
+func getDeployKernelUrl(baremetalConfig BaremetalProvisioningConfig) string {
+	if baremetalConfig.ProvisioningIp != "" {
+		return baremetalConfig.ProvisioningIp + ":" + baremetalHttpPort + baremetalKernelUrlSubPath
```

Comment:
Looks like this is missing http. You can't construct a URL like this, it won't work with IPv6 as the ProvisioningIP needs to be enclosed by brackets `[]`, which go has a helper for: https://golang.org/pkg/net/#JoinHostPort

Try something like this:

```go
fmt.Sprintf("http://%s/%s", net.JoinHostPort(baremetalConfig.ProvisioningIp, baremetalHttpPort), bareme.  SeetalKernelUrlSubPath)
```

Example: https://play.golang.org/p/ffMDzc73p_i

Reply:
Done.